### PR TITLE
prov/psm2: Add shared AV support

### PR DESCRIPTION
**Pull request by j-xiong**

### CODE REVIEW

**Comment by shefty** [comment]
Does this series have any impact on mixing library versions or wire compatibility?

**Comment by shefty** [comment]
Does anything check for this change, to prevent incorrect mixing of library versions?  I didn't notice a version field in the structures being changed, but maybe it is in another part of the code?  Do you need to define new cmds, or are we okay with breaking interoperability between library versions?  (Other providers have broken interoperability too, but at least have run time checks to fail communication attempts).

**Comment by j-xiong** [comment]
No, it's not checked. I could update the provider version though so that application can detect the difference.

File: prov/psm2/src/psmx2_av.c [hunk 1]
```diff
@@ -38,26 +38,18 @@
  * SEQ Query REQ:
  *	args[0].u32w0	cmd
```
Comment (shefty) [inline prov/psm2/src/psmx2_av.c]: Is it possible to insert a 4 or 8 bit version field at the start of the cmd?

**Comment by j-xiong** [comment]
Updated this commit to including version checking for the SEP query protocol:

> prov/psm2: Simplify the protocol for querying scalable endpoint info

**Comment by shefty** [comment]
Thanks!

**Pull request merged**